tests/storage_tests.cpp

```diff
@@ -268,3 +268,184 @@ TEST_CASE("non-unique DBOs") {
                                       make_column("time", &Record::setTime, &Record::time)));
     db.sync_schema();
 }
+
+TEST_CASE("insert") {
```

Comment:
I moved all remaining tests from `tests2.cpp` here cause they are related to `storage_t`